server/app/views/applicant/ProgramCardsSectionParamsFactory.java

```diff
@@ -94,17 +94,24 @@ public ImmutableList<ProgramCardParams> getCards(
       ProgramDefinition program = programDatum.program();
 
       boolean isGuest = personalInfo.getType() == GUEST;
-      String actionUrl = applicantRoutes.review(profile, applicantId, program.id()).url();
+      String actionUrl =
+          profile.isPresent() && applicantId.isPresent()
+              ? applicantRoutes.review(profile.get(), applicantId.get(), program.id()).url()
+              : applicantRoutes.review(program.id()).url();
 
       // Note this doesn't yet manage markdown, links and appropriate aria labels for links, and
       // whatever else our current cards do.
+      String detailsUrl = program.externalLink();
+      if (detailsUrl.isEmpty() || detailsUrl.isBlank()) {
+        detailsUrl =
+            profile.isPresent() && applicantId.isPresent()
```

Comment:
suggestion (non-blocking): You do this same ternary check for `profile.isPresent() && applicantId.isPresent()` in a couple of different places. You could factor it out into a well named method for clarity and ease of reference, but I don't feel strongly about it.